features/query/successful_query/feature.go

```diff
@@ -0,0 +1,51 @@
+package successful_query
+
+import (
+	"context"
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+)
+
+var Feature = harness.Feature{
+	Workflows: Workflow,
+	CheckResult: func(ctx context.Context, r *harness.Runner, run client.WorkflowRun) error {
+		q1, err := r.Client.QueryWorkflow(ctx, run.GetID(), run.GetRunID(), "counterQ")
+		r.Assert.Nil(err)
+		r.Assert.Equal(0, q1)
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "counterInc", nil)
+		r.Assert.Nil(err)
+		q2, err := r.Client.QueryWorkflow(ctx, run.GetID(), run.GetRunID(), "counterQ")
+		r.Assert.Nil(err)
+		r.Assert.Equal(1, q2)
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "counterInc", nil)
+		r.Assert.Nil(err)
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "counterInc", nil)
+		r.Assert.Nil(err)
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "counterInc", nil)
+		r.Assert.Nil(err)
+		q3, err := r.Client.QueryWorkflow(ctx, run.GetID(), run.GetRunID(), "counterQ")
+		r.Assert.Nil(err)
+		// TODO: Something isn't right with go runner. This should fail and it does not
```

Comment:
Ah, so this is a bit weird with how the `Assertion` library works. You have to:

```go
if !r.Assert.Equal(50, q3) {
  return r.LastAssertErr
}
```

Alternatively you could use `r.Require.Equal(50, q3)` which panics

Reply:
:cry: